__tests__/client/prerender.spec.js

```diff
@@ -175,3 +179,55 @@ describe('moveHelmetScripts', () => {
     expect(NodeList.prototype.forEach).not.toHaveBeenCalled();
   });
 });
+
+describe('loadPWA', () => {
+  let error;
+
+  beforeAll(() => {
+    ({ error } = console);
+    console.error = jest.fn();
+    jest.spyOn(window, 'addEventListener');
+    jest.spyOn(window, 'removeEventListener');
+  });
+
+  afterAll(() => {
+    console.error = error;
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call initializePWA on window load and remove the load listener when resolved', async () => {
+    expect.assertions(4);
+
+    const loadPromise = loadPWA();
+    const loadEventHandler = window.addEventListener.mock.calls[0][1];
+
+    loadEventHandler();
+
+    await expect(loadPromise).resolves.toBeUndefined();
+
+    expect(initializePWA).toHaveBeenCalledTimes(1);
+    expect(window.addEventListener).toHaveBeenCalledTimes(1);
+    expect(window.removeEventListener).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not crash the application on failure and remove window load listener', async () => {
```

Comment:
what assertion shows that the application did not crash?

Reply:
Wouldn't catching the Promise ensure that if this function is called will not disrupt wherever it's invoked?

Reply:
I've added a test case to make sure `loadPWA` does not reject if `initializePWA` does fail.